pr_consistency/1.get_merged_prs.py

```diff
@@ -1,116 +1,105 @@
-else:
-    # With an empty JSON file, we use the date of PR#1
-    LAST_MODIFIED_DATE = datetime(2014, 6, 13, 1, 48, 44)
+pull_requests = {}
 
-print("Last modified date: {0}".format(LAST_MODIFIED_DATE))
+try:
 
-# We enclose the following code in a try...finally so that if the updating
-# crashes, we still write out what we had.
+    while True:
 
-try:
+        print('cursor:', cursor)
 
-    # Find the maximum issue number
-    max_issues = repo.get_issues(state='closed', sort='created')[0].number
+        if cursor is None:
+            after = ''
+        else:
+            after = f', after:"{cursor}"'
 
-    # repo.get_pulls doesn't seem to get quite all available pull requests
-    # because state:closed seems to miss some PRs that appear with is:closed
-    # so instead we use get_issues and then get the PR objects manually.
-    # This allows us to also only iterate over issues that have been updated
-    # since a certain time.
-    for i, issue in enumerate(repo.get_issues(since=LAST_MODIFIED_DATE, state='closed')):
+        query = QUERY_TEMPLATE.format(owner=OWNER, repository=NAME, after=after)
 
-        if issue.pull_request is None:
-            continue
+        request = requests.post('https://api.github.com/graphql', json={'query': query}, headers=headers)
 
-        pr = repo.get_pull(issue.number)
+        if request.status_code != 200:
+            raise Exception("Query failed")
 
-        if not pr.merged:
-            continue
+        entries = request.json()['data']['repository']['pullRequests']['edges']
 
-        if pr.base.ref != 'master':
-            continue
+        for entry in entries:
 
-        if str(pr.number) in pull_requests:
-            if pr.updated_at > parse_isoformat(pull_requests[str(pr.number)]['updated']):
-                print("Updating entry for pull request #{} ({} of max {})".format(pr.number, i, max_issues))
-            else:
-                print("Entry for pull request #{} is up to date".format(pr.number, i, max_issues))
-        else:
-            print("Fetching new entry for pull request #{} ({} of max {})".format(pr.number, i, max_issues))
+            pr = entry['node']
 
-        if pr.milestone is None:
-            milestone = None
-        else:
-            milestone = pr.milestone.title
-
-        # Get labels
-        issue = repo.get_issue(pr.number)
-        labels = [label.name for label in issue.labels]
-
-        pull_requests[str(pr.number)] = {'milestone': milestone,
-                                         'title': pr.title,
-                                         'labels': labels,
-                                         'merged': pr.merged_at.isoformat(),
-                                         'updated': pr.updated_at.isoformat(),
-                                         'created': pr.created_at.isoformat(),
-                                         'merge_commit': pr.merge_commit_sha}
+            if pr['number'] == 1041:
```

Comment:
why is 1041 special?

Reply:
not clear to me - maybe @astrofrog has insight? (as the original author here)

Reply:
Leftover debug statement!

Reply:
Will remove!